### PR TITLE
Handle the dependency update due to spring boot 2.3.0-RELEASE

### PR DESCRIPTION
**Pull request by andxu**

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/maven/SpringCloudDependencyManager.java [hunk 1]
```diff
@@ -84,51 +82,53 @@ public boolean update(File file, List<DependencyArtifact> des) throws IOExceptio
...
-                targetVer = getCompatibleVersionWithBootVersion(latestVersions, "2.1.");
-            } else {
-                throw new AzureExecutionException("Unsupported spring-boot version: " + springBootVer);
-            }
-            if (StringUtils.isEmpty(targetVer)) {
-                throw new AzureExecutionException(String.format("Cannot get compatible version for %s:%s with Spring Boot with version %s",
-                        dependency.getGroupId(), dependency.getArtifactId(), springBootVer));
+            String targetVersionText = getCompatibleVersionWithBootVersion(latestVersions, springBootVersion);
+            if (StringUtils.isEmpty(targetVersionText)) {
+                if (isGreaterOrEqualVersion(springBootVersionStr, "2.3.0.RELEASE") && !latestVersions.isEmpty()) {
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/maven/SpringCloudDependencyManager.java:96]: make "2.3.0.RELEASE" constant?

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java [hunk 2]
```diff
@@ -79,87 +81,118 @@ public boolean onActionPerformed(@NotNull AnActionEvent event, @Nullable Operati
...
+                                    "Azure Spring Cloud dependencies are added to your project successfully.",
+                                    summaryVersionChanges(versionChanges));
+                        }
+                    });
+                } catch (DocumentException | IOException | AzureExecutionException | MavenProcessCanceledException e) {
+                    PluginUtil.showErrorNotification("Error",
+                            "Failed to update Azure Spring Cloud dependencies due to error: "
+                                    + e.getMessage());
+                }
+            });
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java:195]: can this method be made simpler by extracting some well-named methods.

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/maven/SpringCloudDependencyManager.java (hunk 1 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/maven/SpringCloudDependencyManager.java]: fixed

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java (hunk 2 above)
Comment (andxu) [reply PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java:160]: done

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java [hunk 3]
```diff
@@ -79,87 +81,83 @@ public boolean onActionPerformed(@NotNull AnActionEvent event, @Nullable Operati
...
-                if (!manager.update(pomFile, versionChanges)) {
-                    PluginUtil.showInfoNotificationProject(project, "Your project is update-to-date.",
-                                                           "No updates are needed.");
+                try {
+                    if (isDirty.get().booleanValue()) {
+                        projectsManager.forceUpdateProjects(Collections.singletonList(mavenProject)).get();
+                    }
+                } catch (InterruptedException | ExecutionException e) {
+                    PluginUtil.showErrorNotification("Error",
+                            "Failed to update update project due to error: "
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java:113]: two `update`s here

**Review by wangmingliang-ms — APPROVED** [review]

**Pull request merged**